src/explore-education-statistics-admin/src/pages/release/edit-release/content/ReleaseContextActionHelpers.ts

```diff
@@ -0,0 +1,277 @@
+import { KeyStatsFormValues } from '@admin/components/editable/EditableKeyStatTile';
+import permissionsService from '@admin/services/permissions/service';
+import {
+  EditableContentBlock,
+  ExtendedComment,
+} from '@admin/services/publicationService';
+import { releaseContentService } from '@admin/services/release/edit-release/content/service';
+import {
+  ContentBlockAttachRequest,
+  ContentBlockPostModel,
+} from '@admin/services/release/edit-release/content/types';
+import { Dictionary } from '@admin/types';
+import {
+  AbstractRelease,
+  ContentSection,
+} from '@common/services/publicationService';
+import { ContentSectionKeys } from './ReleaseContextActionTypes';
+import { useReleaseDispatch } from './ReleaseContext';
+
+const contentSectionComments = (
+  contentSection?: ContentSection<EditableContentBlock>,
+) => {
+  if (contentSection?.content?.length) {
+    return contentSection.content.reduce<ExtendedComment[]>(
+      (allCommentsForSection, content) => {
+        content.comments.forEach(comment =>
+          allCommentsForSection.push(comment),
+        );
+        return allCommentsForSection;
+      },
+      [],
+    );
+  }
+
+  return [];
+};
+
+const getUnresolveComments = (release: AbstractRelease<EditableContentBlock>) =>
+  [
+    ...contentSectionComments(release.summarySection),
+    ...contentSectionComments(release.keyStatisticsSection),
+    ...release.content
+      .filter(_ => _.content !== undefined)
+      .reduce<ExtendedComment[]>(
+        (allComments, contentSection) => [
+          ...allComments,
+          ...contentSectionComments(contentSection),
+        ],
+        [],
+      ),
+  ].filter(comment => comment !== undefined && comment.state === 'open');
+
+export default function useReleaseActions() {
+  const dispatch = useReleaseDispatch();
+
+  async function getReleaseContent(releaseId: string) {
+    dispatch({ type: 'CLEAR_STATE' });
+    const {
+      release,
+      availableDataBlocks,
+    } = await releaseContentService.getContent(releaseId);
+    const canUpdateRelease = await permissionsService.canUpdateRelease(
+      releaseId,
+    );
+    dispatch({
+      type: 'SET_STATE',
+      payload: {
+        unresolvedComments: getUnresolveComments(release),
+        release,
+        availableDataBlocks,
+        canUpdateRelease,
+      },
+    });
+  }
+
+  async function updateAvailableDataBlocks(releaseId: string) {
+    const availableDataBlocks = await releaseContentService.getAvailableDataBlocks(
+      releaseId,
+    );
+    dispatch({
+      type: 'SET_AVAILABLE_DATABLOCKS',
+      payload: { availableDataBlocks },
+    });
+  }
+
+  async function deleteContentSectionBlock(
+    releaseId: string,
```

Comment:
These function signatures aren't great. We can easily get the wrong variable in the wrong place as the param types are mostly strings.

We can improve this significantly by wrapping all of the functions params into a single object bag like props for components:

```typescript
 async function deleteContentSectionBlock(paams: {
    releaseId: string,
    sectionId: string,
}) {
  const { releaseId, sectionId } = params;
  // ...
```

This way, at the call sites, we are way less likely to make a mistake, and we can put the variables in any order in the params object, as long as the names match.

Let's do this for all of these functions to be consistent.